tests/unit/test_crud.py

```diff
@@ -186,6 +192,136 @@ def test_get_contact_by_email_id_miss(dbsession):
     assert contact is None
 
 
+def test_schedule_then_get_acoustic_records_before_time(
+    dbsession, example_contact, maximal_contact, minimal_contact
+):
+    contact_list = [example_contact, maximal_contact, minimal_contact]
+    end_time = datetime.now(timezone.utc) + timedelta(hours=12)
+
+    with StatementWatcher(dbsession.connection()) as watcher:
+        for record in contact_list:
+            schedule_acoustic_record(dbsession, record.email.email_id)
+
+        dbsession.flush()
+
+        record_list = get_all_acoustic_records_before(
+            dbsession,
+            end_time=end_time,
+        )
+        dbsession.flush()
+
+    print(watcher.statements)
+    assert watcher.count == 4
```

Comment:
Nit: ``StatementWatcher`` is more useful when there is a single CRUD function being watched, or if want to check if lazy-loading is unexpectedly being used. I'm guessing there are:

* 3 SQL calls for the 3 calls to ``schedule_acoustic_record``
* 1 SQL call for ``get_all_acoustic_records_before``

The 3 insert calls are uninteresting, I wouldn't use ``StatementWatcher`` to monitor them.

I _am_ curious if accessing ``record.email`` makes a SQL call. But, that is either OK or not, depending how and when we want to get the contact data, which is a decision for the acoustic loop later.